pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -119,13 +119,16 @@ func setupFilterChains(authnPolicy *authn.Policy, sdsUdsPath string, enableSdsTo
 			},
 		}
 	} else {
-		tls.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_ValidationContextSdsSecretConfig{
-			ValidationContextSdsSecretConfig: model.ConstructSdsSecretConfig(model.SDSRootResourceName, sdsUdsPath, model.K8sSAJwtTokenFileName, enableSdsTokenMount),
-		}
-
 		tls.CommonTlsContext.TlsCertificateSdsSecretConfigs = []*auth.SdsSecretConfig{
 			model.ConstructSdsSecretConfig(model.SDSDefaultResourceName, sdsUdsPath, model.K8sSAJwtTokenFileName, enableSdsTokenMount),
 		}
+
+		tls.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_CombinedValidationContext{
+			CombinedValidationContext: &auth.CommonTlsContext_CombinedCertificateValidationContext{
+				DefaultValidationContext:         &auth.CertificateValidationContext{VerifySubjectAltName: []string{} /*subjectAltNames*/},
```

Comment:
Probably remove the comment since 'VerifySubjectAltName' already provides enough info :)

Reply:
Same below

Reply:
I tried yesterday, but for some reason health check fail for istio-proxy in apps without it; need more time to see if it's intermittent or some envoy issues, so put as it is to unblock